alacritty_terminal/src/term/cell.rs

```diff
@@ -92,6 +93,20 @@ impl Cell {
         self.extra.get_or_insert_with(Default::default).zerowidth.push(c);
     }
 
+    #[inline]
+    pub fn underline_color(&self) -> Option<Color> {
+        self.extra.as_ref().and_then(|extra| extra.underline_color)
+    }
+
+    #[inline]
+    pub fn set_underline_color(&mut self, color: Option<Color>) {
```

Comment:
We might want to document how this function works. Passing `None` to reset might seem obvious once you look into it, but it could confuse a first reader.